src/net.py

```diff
@@ -588,10 +606,16 @@ def train(self, epochs, train_x, train_y, val_x, val_y,
                         )
                         # print ('to {}'.format(self.learning_rate))
                     self.minibatch_iteration += 1
+
                 train_error, train_accuracy = self.validator(train_x, train_y)
                 validation_error, validation_accuracy = self.validator(val_x,
                                                                        val_y)
 
+                if self.stopping_rule == 'best_validation' and validation_error < best_error:
```

Comment:
I think there should be a heuristic for determining early stopping by looking at the general trend of the validation error and see if it starts to further itself from the training error.

Reply:
I've been looking for something relatively principled, but yeah.... People seem to run for a few iterations after hitting a local minima, but the value of "few" seems to be pulled out of thin air. Can add something in.

Reply:
if you include some variable, epsilon, that acts as a sensitivity threshold for how much the validation needs to deviate from the training in order to stop that would make it more general purpose. thoughts?